File: Mobile App/Apollo Deliver/lib/Models/PaymentMethod.dart
import 'package:apollodeliver/Models/ShopOrder.dart';
import 'package:apollodeliver/Models/User.dart';

class PaymentMethod {
  int? id;
  int? cartNumber;
  String? nameOnCard;
  String? expirationDate;
  bool? defaultPayment;
  Set<ShopOrder>? shopOrderSet;
  User? user;

  PaymentMethod({
    this.id,
    this.cartNumber,
    this.nameOnCard,
    this.expirationDate,
    this.defaultPayment,
    this.shopOrderSet,
    this.user,
  });

  factory PaymentMethod.fromJson(Map<String, dynamic> json) {
    return PaymentMethod(
      id: json['id'],
      cartNumber: json['cartNumber'],
      nameOnCard: json['nameOnCard'],
      expirationDate: json['expirationDate'],
      defaultPayment: json['defaultPayment'],
      shopOrderSet: (json['shopOrderSet'] as List<dynamic>?)
          ?.map((item) => ShopOrder.fromJson(item as Map<String, dynamic>))
          .toSet(),
      user: json['user'] != null ? User.fromJson(json['user']) : null,
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'cartNumber': cartNumber,
    'nameOnCard': nameOnCard,
    'expirationDate': expirationDate,
    'defaultPayment': defaultPayment,
    'shopOrderSet': shopOrderSet?.map((order) => order.toJson()).toList(),
    'user': user?.toJson(),
  };
}

File: API/apollo_api_services/src/main/java/com/apollo/entity/PaymentMethod.java
package com.apollo.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.Set;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Table(name = "payment_method")
public class PaymentMethod {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private int cartNumber;

    private String nameOnCard;

    private String expirationDate;

    private Boolean defaultPayment;

    @OneToMany(mappedBy = "paymentMethod")
    @JsonBackReference(value = "paymentMethod_shopOrder")
    private Set<ShopOrder> shopOrderSet;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable=false)
    @JsonBackReference(value = "user_paymentMethod")
    private User user;
}
